slack-lint-checks/src/main/java/slack/lint/rx/RxObservableEmitDetector.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2025 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.rx
+
+import com.android.tools.lint.checks.DataFlowAnalyzer
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.ULambdaExpression
+import slack.lint.util.sourceImplementation
+
+class RxObservableEmitDetector : Detector(), SourceCodeScanner {
+  override fun getApplicableUastTypes() = listOf(UCallExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) {
+        val issue = functionToIssue[node.methodName] ?: return
+        val lambdaExpression =
+          requireNotNull(node.valueArguments.lastOrNull() as? ULambdaExpression) {
+            "Missing trailing lambda!"
+          }
+
+        var sendCalled = false
+
+        val visitor =
+          object : DataFlowAnalyzer(emptySet()) {
+            override fun visitCallExpression(node: UCallExpression): Boolean {
+              if (node.methodIdentifier?.name in REQUIRE_ONE_OF) {
+                sendCalled = true
+              }
+              return super.visitCallExpression(node)
+            }
+          }
```

Comment:
While this approach works for the tests I wrote, it isn't very precise: I'm just checking that a method named `send` is called, not that it is invoking on a `ProducerScope` instance. Obviously this isn't ideal so I'm open to suggestions here.

I couldn't find the `ProducerScope` no matter what I tried so I'm clearly missing something

Reply:
Another thing to note: my understanding of `DataFlowAnalyzer` is that I need to pass the node representing the `ProducerScope` to its constructor. It will then invoke the `resolver` method whenever something acts on that node (ie. the code calls `send`). Seeing as I couldn't find the producer I'm passing an empty set. I'm sure there's a better way.

In fact, `TargetMethodDataFlowAnalyzer` seems like it would be useful for this exact use case if I could find the receiver for `send`

Reply:
The ProducerScope should be the first value argument to the call expression, or its extension receiver parameter (if there's an API for that). What did you try?

Reply:
yup, got it. Thanks!

Reply:
Turns out I forgot to import `ProducerScope` in my tests which meant the source scanner didn't know what it was. I missed that it was an error. Once that was fixed I was able to grab the scope from the list of value args

Reply:
https://github.com/slackhq/slack-lints/pull/375/commits/510d836cee3de30b77187450e07f826c3d56293f